docs/services/gcp-services-guide.md

```diff
@@ -0,0 +1,536 @@
+
+#### Connecting MySQL Instance and GKE Cluster
+Before we install Wordpress, we need establish connectivity between our MySQL
+database and our GKE cluster. We can do this by creating a [Private Service 
+Connection](https://cloud.google.com/vpc/docs/configure-private-services-access).
+
+You can create it following the link above or use Crossplane to do it:
+
+* Create a `GlobalAddress` and `Connection` resources:
+```bash
+cat > network.yaml <<EOF
+---
+# example-globaladdress defines the IP range that will be allocated for cloud services connecting
+# to the instances in the given Network.
+apiVersion: compute.gcp.crossplane.io/v1alpha2
+kind: GlobalAddress
+metadata:
+  name: example-globaladdress
+  namespace: $INFRA_NAMESPACE
+spec:
+  providerRef:
+    name: gcp-provider
+    namespace: $INFRA_NAMESPACE
+  reclaimPolicy: Delete
+  name: example-globaladdress
+  purpose: VPC_PEERING
+  addressType: INTERNAL
+  prefixLength: 16
+  network: projects/$PROJECT_ID/global/networks/$NETWORK_NAME
+---
+# example-connection is what allows cloud services to use the allocated GlobalAddress for communication. Behind
+# the scenes, it creates a VPC peering to the network that those service instances actually live.
+apiVersion: servicenetworking.gcp.crossplane.io/v1alpha2
+kind: Connection
+metadata:
+  name: example-connection
+  namespace: $INFRA_NAMESPACE
+spec:
+  providerRef:
+    name: gcp-provider
+    namespace: $INFRA_NAMESPACE
+  reclaimPolicy: Delete
+  parent: services/servicenetworking.googleapis.com
+  network: projects/$PROJECT_ID/global/networks/$NETWORK_NAME
+  reservedPeeringRanges:
+    - example-globaladdress
+EOF
+
+kubectl apply -f network.yaml
+```
+
+* You can verify creation with the following command and output:
+
+*Command*
+```bash
+kubectl -n $INFRA_NAMESPACE get connection example-connection -o custom-columns='NAME:.metadata.name,FIRST_CONDITION:.status.conditions[0].status,SECOND_CONDITION:.status.conditions[1].status'
+```
+*Output*
+```
+NAME                 FIRST_CONDITION   SECOND_CONDITION
+example-connection   True              True
+```
+Wait for both conditions to be true to continue.
+
+## Install Wordpress
+Installing Wordpress requires creating a Kubernetes `Deployment` and load
+balancer `Service`. We will point the deployment to the `wordpressmysql` secret
+that we specified in our claim above for the Wordpress container environment
+variables. It should have been populated with our MySQL connection details after
+the claim became `Bound`.
+
+* Check to make sure `wordpressmysql` exists and is populated:
+
+*Command*
+```bash
+kubectl describe secret wordpressmysql -n app-project1-dev
+```
+
+*Output*
+```bash
+Name:         wordpressmysql
+Namespace:    app-project1-dev
+Labels:       <none>
+Annotations:  <none>
+
+Type:  Opaque
+
+Data
+====
+endpoint:  75 bytes
+password:  27 bytes
+username:  58 bytes
+```
+
+* Define the `Deployment` and `Service` in `wordpress-app.yaml`:
```

Comment:
This line talks about `wordpress-app.yaml`, but the example code uses `wordpress.yaml`.